src/lib/table/table-data-source.ts

```diff
@@ -104,7 +104,15 @@ export class MatTableDataSource<T> extends DataSource<T> {
   sortingDataAccessor: ((data: T, sortHeaderId: string) => string|number) =
       (data: T, sortHeaderId: string): string|number => {
     const value: any = data[sortHeaderId];
-    return _isNumberValue(value) ? Number(value) : value;
+
+    if (_isNumberValue(value)) {
+      const numberValue = Number(value);
+
+      // Numbers beyond `MAX_SAFE_INTEGER` can't be compared reliably so we leave them as strings.
```

Comment:
Is there more explanation for why they aren't compared reliably as floating point numbers?

Reply:
I've added a link to MDN which has an explanation.